src/main/java/org/isf/medicalinventory/manager/MedicalInventoryManager.java

```diff
@@ -267,21 +274,25 @@ public void validateMedicalInventoryRow(MedicalInventory inventory, List<Medical
 				movs.addAll(movBrowserManager.getMovements(medical.getCode(), null, null, null, movFrom, movTo, null, null, null, null));
 			}
 		}
-
+		// Get all the lot of the movements
+		List<Lot> lotOfMovements = movs.stream().map(m -> m.getLot()).collect(Collectors.toList());
+		// Remove duplicates by converting the list to a set
+		Set<Lot> uniqueLots = new HashSet<>(lotOfMovements);
+		// Convert the set back to a list
+		List<Lot> uniqueLotList = new ArrayList<>(uniqueLots);
 		// Cycle fetched movements to see if they impact inventoryRowSearchList
-		for (Movement mov : movs) {
-			Lot movLot = mov.getLot();
-			String lotCodeOfMovement = movLot.getCode();
-			String lotExpiringDate = TimeTools.formatDateTime(movLot.getDueDate(), TimeTools.DD_MM_YYYY);
+		for (Lot lot : uniqueLotList) {
+			String lotCodeOfMovement = lot.getCode();
+			String lotExpiringDate = TimeTools.formatDateTime(lot.getDueDate(), TimeTools.DD_MM_YYYY);
 			String lotInfo = GeneralData.AUTOMATICLOT_IN ? lotExpiringDate : lotCodeOfMovement;
-			Medical medical = mov.getMedical();
+			Medical medical = lot.getMedical();
 			String medicalDesc = medical.getDescription();
 			Integer medicalCode = medical.getCode();
 
 			// Fetch also empty lots because some movements may have discharged them completely
-			Optional<Lot> lot = movStockInsertingManager.getLotByMedical(medical, false).stream().filter(l -> l.getCode().equals(lotCodeOfMovement))
+			Optional<Lot> optLot = movStockInsertingManager.getLotByMedical(medical, false).stream().filter(l -> l.getCode().equals(lotCodeOfMovement))
 							.findFirst();
-			double mainStoreQty = lot.get().getMainStoreQuantity();
+			double mainStoreQty = optLot.get().getMainStoreQuantity();
```

Comment:
given that line 293 has `Optional<Lot> optLog` implies to me that it could be missing.  So why isn't there a test like `optLot.isPresent()` or `!optLog.isEmpty()`?   This is also true before line 492